src/server/api/mutations/updateQuestionResponses.js

```diff
@@ -0,0 +1,115 @@
+import { log } from "../../../lib";
+import { assignmentRequiredOrAdminRole } from "../errors";
+import { cacheableData } from "../../models";
+const ActionHandlers = require("../../../integrations/action-handlers");
+
+export const updateQuestionResponses = async (
+  questionResponses,
+  campaignContactId,
+  loaders,
+  user
+) => {
+  let contact = await loaders.campaignContact.load(campaignContactId);
+  const campaign = await loaders.campaign.load(contact.campaign_id);
+  await assignmentRequiredOrAdminRole(
+    user,
+    campaign.organization_id,
+    contact.assignment_id,
+    contact
+  );
+
+  await cacheableData.questionResponse
+    .save(campaignContactId, questionResponses)
+    .catch(err => {
+      log.error(
+        `Error saving updated QuestionResponse for campaignContactID ${campaignContactId} questionResponses ${JSON.stringify(
+          questionResponses
+        )} error ${err}`
+      );
+    });
+
+  contact = await loaders.campaignContact.load(campaignContactId);
```

Comment:
Why is this needed vs. the one on line 12?  contact data doesn't change with changes to question responses, and if it did, you'd need to 'hard refresh' with e.g. cacheableData.campaignContact.load -- because loaders won't clear on its own.

Reply:
I intended this to ensure we're returning the updated contact.   **is that desirable?**

If so, yes, we'll need to add the hard refresh.  I do think this is desirable because if contact is cached we'd want it refreshed so these respohses are retrieved next time someone retrieves the contact.